charts/consul/templates/webhook-cert-manager-clusterrole.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if or .Values.connectInject.enabled .Values.controller.enabled}}
+{{- if (and (or .Values.connectInject.enabled .Values.controller.enabled) (not (and .Values.global.secretsBackend.vault.enabled .Values.global.secretsBackend.vault.consulConnectInjectCARole .Values.global.secretsBackend.vault.connectInject.tlsCert.secretName .Values.global.secretsBackend.vault.connectInject.caCert.secretName .Values.global.secretsBackend.vault.consulControllerCARole .Values.global.secretsBackend.vault.controller.tlsCert.secretName  .Values.global.secretsBackend.vault.controller.caCert.secretName))) }}
```

Comment:
might be worth extracting the second argument to `and` into a variable for readability.